test/unit_tests/braket/aws/test_aws_device.py

```diff
@@ -848,3 +851,105 @@ def test_get_devices_simulators_only(mock_copy_session, aws_session):
+    midday_weekday_device = MockDevice("ONLINE", ("Weekdays", "07:00", "17:00"))
+    midday_weekend_device = MockDevice("ONLINE", ("Weekend", "07:00", "17:00"))
+    evening_everyday_device = MockDevice("ONLINE", ("Everyday", "17:00", "07:00"))
+    evening_weekday_device = MockDevice("ONLINE", ("Weekdays", "17:00", "07:00"))
+    evening_weekend_device = MockDevice("ONLINE", ("Weekend", "17:00", "07:00"))
+
+    monday_device = MockDevice("ONLINE", ("Monday", "07:00", "17:00"))
+    tuesday_device = MockDevice("ONLINE", ("Tuesday", "07:00", "17:00"))
+    wednesday_device = MockDevice("ONLINE", ("Wednesday", "07:00", "17:00"))
+    thursday_device = MockDevice("ONLINE", ("Thursday", "07:00", "17:00"))
+    friday_device = MockDevice("ONLINE", ("Friday", "07:00", "17:00"))
+    saturday_device = MockDevice("ONLINE", ("Saturday", "07:00", "17:00"))
+    sunday_device = MockDevice("ONLINE", ("Sunday", "07:00", "17:00"))
+    monday_friday_device = MockDevice(
+        "ONLINE", ("Monday", "07:00", "17:00"), ("Friday", "07:00", "17:00")
+    )
+
+    test_devices = (
+        midday_everyday_device,
+        midday_weekday_device,
+        midday_weekend_device,
+        evening_everyday_device,
+        evening_weekday_device,
+        evening_weekend_device,
+        monday_device,
+        tuesday_device,
+        wednesday_device,
+        thursday_device,
+        friday_device,
+        saturday_device,
+        sunday_device,
+        monday_friday_device,
+    )
+
+    test_set = (
+        (datetime(2021, 12, 6, 5, 0, 0), (0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 6, 10, 0, 0), (1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1)),
+        (datetime(2021, 12, 6, 20, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 7, 5, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 7, 10, 0, 0), (1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 7, 20, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 8, 5, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 8, 10, 0, 0), (1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 8, 20, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 9, 5, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 9, 10, 0, 0), (1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0)),
+        (datetime(2021, 12, 9, 20, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 10, 5, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 10, 10, 0, 0), (1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1)),
+        (datetime(2021, 12, 10, 20, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
+        (datetime(2021, 12, 11, 5, 0, 0), (0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0)),
```

Comment:
If I'm interpreting this correctly, 12/11/21 is a Saturday (and a weekend), and the evening weekday device is returning True. Should that be the case ?

Reply:
The evening weekday device - if I am interpreting the definition correctly - would start on Friday evening UTC and end Saturday morning.